### PR TITLE
New release

### PR DESCRIPTION
**Pull request by artem-v**

### CODE REVIEW

File: services-gateway-netty/src/main/java/io/scalecube/services/gateway/GatewaySessionHandler.java [hunk 1]
```diff
@@ -21,42 +22,77 @@
...
 
   /**
    * On session close handler.
    *
    * @param session websocket session (not null)
+   * @return mono result
    */
-  default void onSessionClose(GatewaySession session) {
-    LOGGER.info("Session closed: " + session);
+  default Mono<Void> onSessionClose(GatewaySession session) {
```
Comment (segabriel) [inline services-gateway-netty/src/main/java/io/scalecube/services/gateway/GatewaySessionHandler.java]: Why do we need such changes? I mean what does using Mono<Void> give us? If you need to throw en exception you can just throw it, no need to wrap it to `Mono.error`

Comment (artem-v) [reply services-gateway-netty/src/main/java/io/scalecube/services/gateway/GatewaySessionHandler.java]: You 're talking about onSessionClose specifically?

Comment (segabriel) [reply services-gateway-netty/src/main/java/io/scalecube/services/gateway/GatewaySessionHandler.java]: @artem-v yes, it is, and about `onSessionOpen`

Comment (artem-v) [reply services-gateway-netty/src/main/java/io/scalecube/services/gateway/GatewaySessionHandler.java]: I also was unsure about it and make it to be in sync with onSessionOpen. But to be fair u don't need to be async  on close.

Comment (artem-v) [reply services-gateway-netty/src/main/java/io/scalecube/services/gateway/GatewaySessionHandler.java]: > @artem-v yes, it is, and about `onSessionOpen`

Im planning to do something asynchronous during session-open phase, and don't want to block netty thread, hence this mono approach.

Comment (segabriel) [reply services-gateway-netty/src/main/java/io/scalecube/services/gateway/GatewaySessionHandler.java]: But for `onSessionOpen` is redundant as well

Comment (segabriel) [reply services-gateway-netty/src/main/java/io/scalecube/services/gateway/GatewaySessionHandler.java]: > Im planning to do something asynchronous during session-open phase, and don't want to block netty thread, hence this mono approach.

I thought it's already asynchronous.

Comment (artem-v) [reply services-gateway-netty/src/main/java/io/scalecube/services/gateway/GatewaySessionHandler.java]: Yes
You're correct. 
I will refactor this change.

Comment (segabriel) [reply services-gateway-netty/src/main/java/io/scalecube/services/gateway/GatewaySessionHandler.java]: @artem-v Thank you! My apologies for stopping merge this PR.

**Pull request merged**